### PR TITLE
introduce a new function non_negative_difference

### PR DESCRIPTION
**Pull request by zhexuany**
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: influxql/functions.go [hunk 1]
```diff
@@ -216,6 +218,12 @@ func (r *FloatDifferenceReducer) Emit() []FloatPoint {
 		// Calculate the difference of successive points.
 		value := r.curr.Value - r.prev.Value
 
+		// if its non_negative_difference discard any negative value. Since
```
Comment (jsternberg) [inline influxql/functions.go]: Can you modify this comment to use proper capitalization and fix the typo in `makred` to `marked`?

File: influxql/parser_test.go [hunk 2]
```diff
@@ -314,6 +314,56 @@ func TestParser_ParseStatement(t *testing.T) {
 			},
 		},
 
+		// non_negtative_difference
```
Comment (jsternberg) [inline influxql/parser_test.go]: Another typo here.

File: influxql/functions.go (hunk 1 above)
Comment (zhexuany) [inline influxql/functions.go]: done.

File: influxql/parser_test.go (hunk 2 above)
Comment (zhexuany) [inline influxql/parser_test.go]: done

**Comment by jsternberg** [comment]
Can you rebase to a single commit and update the changelog? Then I'll double check that this is OK to merge and hopefully have it done before the end of the day.

**Comment by zhexuany** [comment]
@jsternberg Done.

**Comment by jsternberg** [comment]
Thanks!

**Pull request merged**